enumerable.rb

```diff
@@ -0,0 +1,106 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    my_arr = []
+    my_each { |i| my_arr.push(i) if yield(i) }
+    my_arr
+  end
+
+  def my_all?(*args)
+    condition = true
+    if !args[0].nil?
```

Comment:
You don't consider the possibility when no block or argument is given, so you always return true even if you have an array of [nil, false].my_all?

Consider the possibility of giving a class as an argument (Integer, String). Right now you are returning false when you should return true.
Consider the possibility of giving a Regex as an argument (/d/). Right now you are returning false when you should return true.
Consider the possibility of giving a number as an argument (%). Right now you are returning false when you should return true.